### ISSUE TITLE
Dataset processing

### ISSUE THREAD

**Issue by hlml** [ahmdtaha/knowledge_evolution#3]
Hi Ahmed, thanks a lot for this interesting work and open codebase! Would it be possible to share the data processing scripts for the other datasets used in the paper?

**Comment by ahmdtaha** [comment]
Hi Hattie,

Thanks for your interest in knowledge evolution. I didn't release the other data loaders because my personal implementation is different. My implementation uses Nvidia Dali, so it is more complex. In contrast, I used PyTorch dataset loaders in this public implementation to promote knowledge evolution.

Can you please share which data processing scripts you want? I will try to implement it and add to the repos. This will take sometime due to my current commitments.

Thanks

**Comment by hlml** [comment]
Thanks Ahmed!

Ah, I see. In that case I can try to implement the other datasets myself.

It would be really helpful if you can give me some high level pointers on any specific processing you did for the other datasets (stanford dogs, mit67, aircraft). In particular if you used any of the metadata files associated with the datasets, that'd be helpful to know for reproduction :).

**Comment by ahmdtaha** [comment]
I didn't use any metadata files with these datasets (Dogs,MIT,Aircrafts), just the class labels. I used the official dataset splits. I will prepare a script for one of these datasets. I want to see how difficult to convert the PytorchFlower loader to another dataset loader. Pick one of these three datasets?

Assuming you are interesting in small datasets, I want to share something with you. I don't think Dogs/MIT/Aircraft are ideal dataset to study overfitting. These datasets have two entangled problems: small size and fine-grained. Accordingly, I wish I had more datasets like Flower: small size and coarse-grained. Basically, it is easier to focus on one problem at a time.

Thanks

**Comment by hlml** [comment]
Ok, thanks, I appreciate it! How about the aircrafts dataset?

Thanks for the thought, that's a good point. Do you think knowledge evolution focuses on the small size problem only, or would it also help with fine-grained classification?

**Comment by ahmdtaha** [comment]
Knowledge evolution supports small, fine-grained, and even large datasets. I evaluated knowledge evolution on ImageNet in the paper appendix. Feel free to check this out. That being said, knowledge evolution is best suited for small coarse grained datasets. Maybe you can figure out a different use for knowledge evolution :)

I will add the aircrafts dataset loader -- hopefully by tomorrow.

**Comment by ahmdtaha** [comment]
Hi Hattie,

I added Aircarfts100 dataset loader. Once I have the lists/*.csv files ready, It took few minutes to add the Aircraft100Pytorch -- just change a single word from Flower102Pytorch. I should add a parent class to these two classes in the future. You will find the lists dir [here](https://drive.google.com/file/d/1lyaro579PRFAIyBZoPpLBjCzKdvsfEH2/view?usp=sharing)

I ran a partial test and everything works normally. If you plan to add more loaders, please feel free to submit a pull request.
If you are serious about knowledge evolution, I recommend you use Nvidia Dali loaders. It will save you a ton of time in the long run.

Thanks

**Comment by hlml** [comment]
Thanks a lot for adding this Ahmed! I will give it a try. Definitely very inspiring work :)
